src/sources/soundsourceffmpeg.cpp

```diff
@@ -119,98 +228,81 @@ SoundSource::OpenResult SoundSourceFFmpeg::tryOpen(const AudioSourceConfig& /*au
+                << "avformat_find_stream_info() failed and returned"
+                << avformat_find_stream_info_result;
         return OpenResult::FAILED;
     }
 
     //debug only (Enable if needed)
     //av_dump_format(m_pFormatCtx, 0, qBAFilename.constData(), false);
 
-    // Find the first audio stream
-    m_iAudioStream = -1;
-
-    for (unsigned int i = 0; i < m_pFormatCtx->nb_streams; i++)
-        if (m_pFormatCtx->streams[i]->codec->codec_type == AVMEDIA_TYPE_AUDIO) {
-            m_iAudioStream = i;
-            break;
-        }
-
-    if (m_iAudioStream == -1) {
-        qDebug() <<
-                 "SoundSourceFFmpeg::tryOpen: cannot find an audio stream: cannot open"
-                 << localFileName;
-        return OpenResult::FAILED;
-    }
-
-    // Get a pointer to the codec context for the audio stream
-    m_pCodecCtx = m_pFormatCtx->streams[m_iAudioStream]->codec;
-
-    // Find the decoder for the audio stream
-    if (!(m_pCodec = avcodec_find_decoder(m_pCodecCtx->codec_id))) {
-        qDebug() << "SoundSourceFFmpeg::tryOpen: cannot find a decoder for" <<
-                localFileName;
+    // Find and open audio stream for decoding
+    AVStream* pAudioStream = findFirstAudioStream(m_pFormatCtx);
+    if (pAudioStream == nullptr) {
+        qWarning() << "[SoundSourceFFmpeg]"
+                << "No audio stream found";
         return OpenResult::UNSUPPORTED_FORMAT;
     }
-
-    if (avcodec_open2(m_pCodecCtx, m_pCodec, nullptr)<0) {
-        qDebug() << "SoundSourceFFmpeg::tryOpen: cannot open" << localFileName;
-        return OpenResult::FAILED;
+    const OpenResult openAudioStreamResult = openAudioStream(pAudioStream);
+    if (openAudioStreamResult != OpenResult::SUCCEEDED) {
+        return openAudioStreamResult; // early exit on any error
     }
+    m_pAudioStream = pAudioStream;
 
-    m_pResample = std::make_unique<EncoderFfmpegResample>(m_pCodecCtx);
-    m_pResample->openMixxx(m_pCodecCtx->sample_fmt, AV_SAMPLE_FMT_FLT);
-
-    setChannelCount(m_pCodecCtx->channels);
-    setSamplingRate(m_pCodecCtx->sample_rate);
-    setFrameCount((qint64)round((double)((double)m_pFormatCtx->duration *
-                                         (double)m_pCodecCtx->sample_rate) / (double)AV_TIME_BASE));
-
-    qDebug() << "SoundSourceFFmpeg::tryOpen: Sampling rate: " << getSamplingRate() <<
-             ", Channels: " <<
-             getChannelCount() << "\n";
-    if (getChannelCount() > 2) {
-        qDebug() << "ffmpeg: No support for more than 2 channels!";
-        return OpenResult::FAILED;
+    const SINT channelCount = mixxx::getChannelCount(m_pAudioStream);
+    if (channelCount > kMaxChannelCount) {
+        qWarning() << "[SoundSourceFFmpeg]"
+                << "Unsupported number of channels:"
+                << channelCount << ">" << kMaxChannelCount;
+        return OpenResult::UNSUPPORTED_FORMAT;
     }
+    const SINT samplingRate = mixxx::getSamplingRate(m_pAudioStream);
```

Comment:
I used the function names from our AudioSource interface, which cause naming 
conflicts that need to be resolved explicitly as already mentioned by 
Daniel. Will be fixed.

Why I introduced those functions: AVCodec (codec) in AVStream will be 
deprecated soon and is replaced by AVCodecParameters (codecparam) in FFmpeg 
3.1. We shouldn't access the member codec everywhere in our code and better 
hide it behind some functions that will make the transition easier.

I'm currently using FFmpeg 3.0 from Fedora 24.

On 09.09.2016 08:28, Tuukka Pasanen wrote:
>
> In src/sources/soundsourceffmpeg.cpp 
> <https://github.com/mixxxdj/mixxx/pull/1003#discussion_r78134076>:
>
> >      }
> > +    const SINT samplingRate = mixxx::getSamplingRate(m_pAudioStream);
>
> Hmm.. is this just from the history?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/1003/files/14399cf62b5d5ee96a4d0d3a361af8ab165b2144#r78134076>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-anKS29Ifs4nUVS465gJwhMakqe5Xks5qoPyOgaJpZM4J2QNu>.
>